sql-parser/src/main/java/io/crate/sql/parser/AstBuilder.java

```diff
@@ -1190,14 +1190,35 @@ public Node visitJoinRelation(SqlBaseParser.JoinRelationContext context) {
                 criteria = new JoinOn((Expression) visit(context.joinCriteria().booleanExpression()));
             } else if (context.joinCriteria().USING() != null) {
                 List<String> columns = identsToStrings(context.joinCriteria().ident());
-                criteria = new JoinUsing(columns);
+                criteria = new JoinUsing(extractQualifiedName(left), extractQualifiedName(right), columns);
             } else {
                 throw new IllegalArgumentException("Unsupported join criteria");
             }
         }
         return new Join(getJoinType(context.joinType()), left, right, Optional.of(criteria));
     }
 
+    private static QualifiedName extractQualifiedName(Relation rel) {
```

Comment:
This instanceof handling in this method might be easy to forget to update in case we add new relations.

I like the solution you had in the first commit better. Was there a reason why you choose to go with this instead?

Reply:
I like the original commit better too.